### ISSUE TITLE
Variable Buffer Size

### ISSUE THREAD

**Issue by timoxd7** [nathandunk/BetterSerialPlotter#19]
Hi, great project!

I sometimes have many datapoints, exceeding the internal Buffer. Would it be possible to add a configurable sized buffer?

Also, a Checkmark with "endless" would be nice so that the software will just keep going as long as you want. Could be done with chunks of allocated heap, when exceeding just add a new chunk.

* With buffer i mean the plot in the scrollable viewport

**Comment by nathandunk** [comment]
Hi @timoxd7, it would definitely be possible to change the buffer size to be configurable. As I am adding more features, I am trying to think of the best way to do this. Because it is related to the serial data coming in, I would likely add it on the top line next to other serial information. I am thinking a button to press which pops open a text box to fill, with a checkmark next to it that says endless. Does that sound like what you are looking for?

**Comment by timoxd7** [comment]
@nathandunk Yes that sounds perfect. If you are on writing things depending on the data input, i also have another small idea maybe worth thinking about. I sometimes do not have the UART interface working, but a file which contains all the potential UART output.

To read the serial in from file, combined with an endless buffer would make it possible to discover recorded data with your great tool with all the serial parsing features and plot navigation.

**Comment by timoxd7** [comment]
@nathandunk that would be perfect.

Another idea with this implemented would be to read serial from file. I have a device not connected to a PC directly but with console to file. Combinet with an endless buffer would make it possible to parse the file with your great tool and use the plot navigations to view all captured data.
